order/src/main/java/org/order/OrderController.java

```diff
@@ -0,0 +1,37 @@
+package org.order;
+
+import java.util.List;
+import lombok.AllArgsConstructor;
+import org.order.dto.PurchaseOrderDto;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@AllArgsConstructor
+public class OrderController {
+
+    private final OrderService orderService;
+
+    @ResponseStatus(HttpStatus.CREATED)
+    @PostMapping("/orders")
```

Comment:
consider that you are expecting only one order, we can change to singular
```suggestion
    @PostMapping("/order")
```

Reply:
from what i heard and read it's a good practice to use plural 🤔 
![image](https://github.com/user-attachments/assets/2b281038-135f-4229-a02b-d3f61057e535)
https://stackoverflow.blog/2020/03/02/best-practices-for-rest-api-design/
https://www.freecodecamp.org/news/rest-api-best-practices-rest-endpoint-design-examples/

Reply:
I would agreed on plural when you have a "dynamic" filter, that you will potentially have more than one result, however since that you have the ID on the path it means that you'll get 0-1 item and never more than that, by that I would prefer to have it singular. Also, thanks to it, I know that it'll be a get for a single one. In general for this particular case it's a bit of preference, my one is go on singular for this case and plural when I know that I'll get a list of it. Also, recently I got one comment at work four our Architect, that makes me think a bit more about these plural names, the example is `/mouse` singular so on plural it should be `/mice`, but we will probably put `/mouses` which doesn't exists, so how to handle it :D . So here I'll leave it up to you, if you also want to have some extra fun with this - which I do recommend - try to use SpringHATEOS into your project https://spring.io/projects/spring-hateoas it'll also give you more example and ideas about how to handle these path endpoint ;)